resonance/linear_systems.py

```diff
@@ -0,0 +1,277 @@
+import collections
+from inspect import getargspec
+
+import numpy as np
+import pandas as pd
+import matplotlib.animation as animation
+
+
+class MeasurmentsDict(collections.MutableMapping, dict):
+
+    def _compute_value(self, key):
+
+        func = self._funcs[key]
+
+        def get_par(k):
+            try:
+                v = self._par[k]
+            except KeyError:
+                try:
+                    v = self._coord[k]
+                except KeyError:
+                    v = self[k]
+            return v
+
+        # TODO : getargspec is deprecated, supposedly signature cna do the same
```

Comment:
As far as I know, it should be reliably ordered: https://docs.python.org/3.6/library/inspect.html#inspect.Signature.parameters

```python
>>> from inspect import signature
>>> signature(lambda a, b: a).parameters
mappingproxy(OrderedDict([('a', <Parameter "a">), ('b', <Parameter "b">)]))
```

Reply:
Ah, it is an ordered dict. Great. I'm thinking an ordered dict is going to be the right choice for storing states, etc. too.